pkg/pillar/cmd/nodeagent/handletimers.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package nodeagent
+
+import (
+	"fmt"
+	"syscall"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zboot"
+	log "github.com/sirupsen/logrus"
+)
+
+// node health timer funcions
+
+// called periodically from agentwatch context
+func handleDeviceTimers(ctx *nodeagentContext) {
```

Comment:
Why did you invent a new pattern where you periodically check? I think the explicit t1 and t2 timers in zedagent is easier to understand instead of depending on the stillRunning timer.

Reply:
sure